docarray/doc_index/backends/weaviate_doc_index.py

```diff
@@ -0,0 +1,525 @@
+                    data_object=parsed_doc,
+                    class_name=index_name,
+                    vector=vector,
+                )
+
+    def _text_search(self, query: str, search_field: str, limit: int) -> _FindResult:
+        index_name = self._db_config.index_name
+        bm25 = {"query": query, "properties": [search_field]}
+
+        results = (
+            self._client.query.get(index_name, self.properties)
+            .with_bm25(bm25)
+            .with_limit(limit)
+            .with_additional(["score", "vector"])
+            .do()
+        )
+
+        return self._format_response(results["data"]["Get"][index_name], "score")
+
+    def _text_search_batched(
+        self, queries: Sequence[str], search_field: str, limit: int
+    ) -> _FindResultBatched:
+        qs = []
+        for i, query in enumerate(queries):
+            bm25 = {"query": query, "properties": [search_field]}
+
+            q = (
+                self._client.query.get(self._db_config.index_name, self.properties)
+                .with_bm25(bm25)
+                .with_limit(limit)
+                .with_additional(["score", "vector"])
+                .with_alias(f'query_{i}')
+            )
+
+            qs.append(q)
+
+        results = self._client.query.multi_get(qs).do()
+
+        docs_and_scores = [
+            self._format_response(result, "score")
+            for result in results["data"]["Get"].values()
+        ]
+
+        docs, scores = zip(*docs_and_scores)
+        return list(docs), list(scores)
+
+    def execute_query(self, query: Any, *args, **kwargs) -> Any:
+        da_class = DocumentArray.__class_getitem__(
+            cast(Type[BaseDocument], self._schema)
+        )
+
+        if isinstance(query, self.QueryBuilder):
+            results = query._query.do()
+            docs = results["data"]["Get"][self._db_config.index_name]
+
+            def f(doc):
+                return self._schema.from_view(self._parse_weaviate_result(doc))
+
+            return da_class([f(doc) for doc in docs])
+
+    def num_docs(self) -> int:
+        index_name = self._db_config.index_name
+        result = self._client.query.aggregate(index_name).with_meta_count().do()
+        # TODO: decorator to check for errors
+        total_docs = result["data"]["Aggregate"][index_name][0]["meta"]["count"]
+
+        return total_docs
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """Map python type to database type."""
+        for allowed_type in WEAVIATE_PY_VEC_TYPES:
+            if issubclass(python_type, allowed_type):
+                return np.ndarray
+
+        if python_type in WEAVIATE_PY_TYPES:
+            return python_type
+
+        raise ValueError(f'Unsupported column type for {type(self)}: {python_type}')
+
+    def build_query(self) -> BaseDocumentIndex.QueryBuilder:
+        return self.QueryBuilder(self)
+
+    class QueryBuilder(BaseDocumentIndex.QueryBuilder):
```

Comment:
What I see here now is that there is no `Hybrid Search` implemented right?

Reply:
yes, that is not implemented yet.